### PR TITLE
SCMOD-4829: Restrict scope change when repository configs exist

### PR DESCRIPTION
**Pull request by davidegph**
https://mc4t01643.itcs.softwaregrp.net/browse/SCMOD-4829

* scope in the global configuration cannot be changed anymore if a repository configuration exists and uses its key
* a tenant configuration cannot be deleted anymore if a repository configuration exists and uses its tenant id
* added 2 tests to verify these behaviors

http://sou-jenkins2.hpeswlab.net/job/CAFDataProcessing/job/CAFDataProcessing%7Edata-processing-service%7ESCMOD-4829%7ECI/

### CODE REVIEW

**Comment by davidegph** [comment]
Hi @dermot-hardy ,

would you mind review this pull request, please?

**Comment by dermot-hardy** [comment]
I'm quite confused about the second point:
> a tenant configuration cannot be deleted anymore if a repository configuration exists and uses its tenant id

I'm not sure that we should be disallowing this.  It's not clear to me why we are.  Was there a conversation about this that I've forgotten?  The Jira only covers the first point which is clearly a bug.  But on this second point surely you could easily get into the same state by adding a repository configuration for a setting that hasn't been set at the tenant level - so I don't understand why we're doing that.

**Review by dermot-hardy — CHANGES_REQUESTED** [review]
I'm quite confused about the second point:
> a tenant configuration cannot be deleted anymore if a repository configuration exists and uses its tenant id

I'm not sure that we should be disallowing this.  It's not clear to me why we are.  Was there a conversation about this that I've forgotten?  The Jira only covers the first point which is clearly a bug.  But on this second point surely you could easily get into the same state by adding a repository configuration for a setting that hasn't been set at the tenant level - so I don't understand why we're doing that.

**Comment by davidegph** [comment]
Hi @dermot-hardy ,

I misunderstood the fact that a tenant configuration should have always been present to have a configuration for a repository. I have removed this part, and now the code only allows to spot the change of scope in the global configuration, as mentioned in the jira.

I managed to [build it](http://sou-jenkins2.hpeswlab.net/job/CAFDataProcessing/job/CAFDataProcessing%7Edata-processing-service%7ESCMOD-4829%7ECI/17/) too.

File: processing-service-container/src/test/java/com/github/cafdataprocessing/processing/service/tests/RepositoryConfigIT.java [hunk 1]
```diff
@@ -227,8 +227,8 @@ public void effectiveChainOfConfigs() throws ApiException
         Assert.assertTrue(configValue3.getValueType().equals(ValueTypeEnum.CUSTOM));
         Assert.assertEquals(configValue3.getValue(), "from_the_repo");
         
-        TENANT_API.deleteTenantConfigs(tenantId);
         REPOSITORY_API.deleteRepositoryConfigs(tenantId, repositoryId);
+        TENANT_API.deleteTenantConfigs(tenantId);
```
Comment (dermot-hardy) [inline processing-service-container/src/test/java/com/github/cafdataprocessing/processing/service/tests/RepositoryConfigIT.java]: Why is this change required?

**Review by dermot-hardy — APPROVED** [review]

**Review by Anthony-McGreevy — APPROVED** [review]

**Pull request merged**